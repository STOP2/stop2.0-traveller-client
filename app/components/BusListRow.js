import React, { Component } from 'react';
import { View } from 'react-native';
import { DefaultText } from '../components/textComponents';
import VehicleImage from '../components/VehicleImage';

import styles from '../styles/stylesheet';

const BusListRow = ({ vehicleType, line, destination, arrival }) => (
          <View style={styles.busrow}>
            <View style={styles.flex1}>
              <VehicleImage style={styles.busrowImage} vehicleType={vehicleType}/>
            </View>
            <DefaultText style={styles.busrowText}>{line}</DefaultText>
            <DefaultText style={styles.busrowTextBlack2}>{destination}</DefaultText>
            <DefaultText style={styles.busrowTextBlack}>{arrival} min</DefaultText>
          </View>
);

BusListRow.propTypes = {
    vehicleType: React.PropTypes.number.isRequired,
    line: React.PropTypes.string.isRequired,
    destination: React.PropTypes.string.isRequired,
    arrival: React.PropTypes.number.isRequired,
};

export default BusListRow;
