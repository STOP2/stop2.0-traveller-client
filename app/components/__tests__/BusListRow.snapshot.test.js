import 'react-native'
import React from 'react'
import BusListRow from '../BusListRow'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'


jest.mock('react-native-localization')

describe('BusListRow component', () =>
{
    it('renders correctly', () =>
    {
        let tree = renderer.create(<BusListRow vehicleType={0} line="506" destination="Pasila" arrival={5} />).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
