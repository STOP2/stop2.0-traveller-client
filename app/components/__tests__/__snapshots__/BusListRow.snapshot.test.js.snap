exports[`BusListRow component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#666666",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 60,
      "padding": 10,
    }
  }>
  <View
    style={
      Object {
        "flex": 1,
      }
    }>
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": 20,
          "marginLeft": 5,
          "width": 20,
        }
      } />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "gotham-rounded-book",
        },
        Object {
          "flex": 1,
          "fontSize": 15,
        },
      ]
    }>
    506
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "gotham-rounded-book",
        },
        Object {
          "color": "black",
          "flex": 3,
          "fontSize": 15,
        },
      ]
    }>
    Pasila
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "gotham-rounded-book",
        },
        Object {
          "color": "black",
          "flex": 1,
          "fontSize": 15,
        },
      ]
    }>
    5
     min
  </Text>
</View>
`;
